src/otx/engine/engine.py

```diff
@@ -0,0 +1,486 @@
+
+        Returns:
+            list | None: The predictions if `return_predictions` is True, otherwise None.
+
+        Example:
+        >>> engine.predict(
+            datamodule=OTXDataModule(),
+            checkpoint="checkpoint.ckpt",
+            return_predictions=True,
+        )
+
+        CLI Usage:
+            1. you can pick a model.
+                ```python
+                otx predict
+                    --model <CONFIG | CLASS_PATH_OR_NAME> --data_root <DATASET_PATH, str>
+                    --checkpoint <CKPT_PATH, str>
+                ```
+            2. If you have a ready configuration file, run it like this.
+                ```python
+                otx predict --config <CONFIG_PATH, str> --checkpoint <CKPT_PATH, str>
+                ```
+        """
+        lit_module = self._build_lightning_module(
+            model=self.model,
+            optimizer=self.optimizer,
+            scheduler=self.scheduler,
+        )
+
+        self._build_trainer(**kwargs)
+
+        return self.trainer.predict(
+            model=lit_module,
+            datamodule=datamodule if datamodule is not None else self.datamodule,
+            ckpt_path=str(checkpoint) if checkpoint is not None else checkpoint,
+            return_predictions=return_predictions,
+        )
+
+    def export(self, *args, **kwargs) -> None:
+        """Export the trained model to OpenVINO Intermediate Representation (IR) or ONNX formats."""
+        raise NotImplementedError
+
+    @classmethod
+    def from_config(cls, config: EngineConfig | str | Path) -> Engine:
+        """Create an instance of the Engine class from a configuration file or object.
+
+        Args:
+            config (EngineConfig | str | Path): The path to the configuration file or the configuration object.
+
+        Returns:
+            Engine: An instance of the Engine class.
+
+        Raises:
+            FileNotFoundError: If the configuration file is not found.
+            TypeError: If the configuration object is not of type EngineConfig.
+
+        Example:
+        >>> runner = Engine.from_config(
+            config="config.yaml",
+        )
+        """
+        model = None
+        data_root = ""
+        datamodule = None
+        if isinstance(config, (str, Path)):
+            with Path(config).open() as f:
+                config_dict = yaml.safe_load(f)
+            data_config = config_dict.pop("data", {})
+            datamodule = OTXDataModule.from_config(config=data_config)
+            model = instantiate_class(args=(), init=config_dict.pop("model", {}))
+            optimizer = partial_instantiate_class(init=config_dict.pop("optimizer", {}))
+            scheduler = partial_instantiate_class(init=config_dict.pop("scheduler", {}))
+            engine_args = config_dict.pop("engine", config_dict)
+            engine_config = EngineConfig(**engine_args)
+        else:
+            engine_config = config
+
+        data_root = config_dict.pop("data_root", "")
```

Comment:
Is this correct? What happens if `config` is not of instance `str | Path`? In that case `config_dict` is unassigned.